aws/cognito_user_pool/variables.tf

```diff
@@ -74,3 +74,20 @@ variable "groups" {
   ))
   default = {}
 }
+
+variable "create_user_messaging" {
+  description = "Provides basic customized messaging for new accounts"
+  type = object(
+    {
+      email_subject = optional(string)
+      email_message = optional(string)
+      sms_message   = optional(string)
+    }
+  )
+}
+
+variable "allow_admin_create_user_only" {
```

Comment:
I'm not sure if there's a better name for this, so I just used the one that the aws module uses. I'd opt for something like "open_registration" but I don't know if that's an appropriate substitution

Reply:
yea, i think just using the `aws` module's naming for it is fine